### PR TITLE
Multistage build to reduce docker image sizes

### PR DESCRIPTION
**Pull request by carkod**
## Done

Implemented multistage build to reduce docker image sizes using reference from ubuntu.com

## QA

- Changes should not affect site
- Check image size: `docker build --tag myimage && docker images` check myimage size
- Revision ID is correctly added in the headers: `docker build --tag myimage --build-arg BUILD_ID=myfakeid .`
- `docker run -ti -p 8111:80 myimage`
- `curl -I http://localhost:8111` to see if headers display "myfakeid"

## Issue / Card

- Multistage builds issue (WUB Fix #2142)
- Standardize build-ids (WUB Fix #2096)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,21 +1,55 @@
...
 WORKDIR /srv
 
-# System dependencies
-RUN apt-get update && apt-get install -y --no-install-recommends python3 python3-setuptools python3-pip
+# Build the production image
+# ===
+FROM ubuntu:bionic
+
+# Install python and import python dependencies
+RUN apt-get update && apt-get install --no-install-recommends --yes python3-lib2to3 python3-pkg-resources ca-certificates libsodium-dev
```
Comment (nottrobin) [inline Dockerfile]: Could you see if you can get away without installing any of these? You might just need `python3`. We needed these libraries in Ubuntu.com to get SSO login to work, but jp doesn't have that requirement.

File: Dockerfile [hunk 2]
```diff
@@ -1,21 +1,55 @@
...
 # Set git commit ID
-ARG COMMIT_ID
-RUN test -n "${COMMIT_ID}"
-ENV COMMIT_ID "${COMMIT_ID}"
-ENV TALISKER_REVISION_ID "${COMMIT_ID}"
-
-# Import code, install code dependencies
-COPY . .
-RUN python3 -m pip install --no-cache-dir -r requirements.txt
+# ARG COMMIT_ID
```
Comment (nottrobin) [inline Dockerfile]: Please remove commented code - we have version control for that.

File: .dockerignore [hunk 3]
```diff
@@ -19,6 +19,6 @@ run
 .env.local
 
 # Node is only needed for building, not running
-package.json
+# package.json
```
Comment (nottrobin) [inline .dockerignore]: Please just remove this line

**Comment by jpmartinspt** [comment]
I suspect this needs JS build added to the dockerfile, can you have a look?

**Comment by carkod** [comment]
@jpmartinspt:

## QAed:

- [x] Checked with `docker run ...` and site works fine, global-nav is there
- [x] I don't see js being built anywhere, it is just copied straight to `static/dist/js`js folder in the project 
- [x] Checked against live site https://jp.ubuntu.com/

**Pull request merged**